// The Firmament project
// Copyright (c) 2013 Ionel Gog <ionel.gog@cl.cam.ac.uk>
// Copyright (c) 2013 Malte Schwarzkopf <malte.schwarzkopf@cl.cam.ac.uk>
//
// Flow graph node type protobuf.

package firmament;

message FlowNodeType {

  enum NodeType {
    ROOT_TASK = 0;
    SCHEDULED_TASK = 1;
    UNSCHEDULED_TASK = 2;
    GLOBAL_AGGREGATOR = 3;
    JOB_AGGREGATOR = 4;
    RACK = 5;
    MACHINE = 6;
    PU = 7;
    SINK = 8;
    UNKNOWN = 9;
  }

  required NodeType type = 1 [default = UNKNOWN];
}
